### PR TITLE
Tweak: Add menus to the I/E [ED-6430]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: core/app/modules/import-export/import.php [hunk 1]
```diff
@@ -59,4 +86,48 @@ private function init_adapters( array $manifest_data ) {
...
+	private function save_elements_of_imported_posts() {
+		foreach ( $this->documents_elements as $new_id => $document_elements ) {
+			$document = Plugin::$instance->documents->get( $new_id );
+			$updated_elements = $document->on_import_replace_dynamic_content( $document_elements, $this->map_old_new_post_ids );
+			$document->save( [ 'elements' => $updated_elements ] );
+		}
+	}
+
+	private function update_object_id_of_imported_menu_items() {
+		foreach ( $this->map_old_new_post_ids as $old_post_id => $new_post_id ) {
```
Comment (yossi-elementor) [inline core/app/modules/import-export/import.php:99]: Remove `old_post_id`

File: core/app/modules/import-export/import.php [hunk 2]
```diff
@@ -59,4 +86,48 @@ private function init_adapters( array $manifest_data ) {
...
+				continue;
+			}
+
+			$menu_item_type = get_post_meta( $new_post_id, '_menu_item_type' )[0];
+
+			if( 'post_type' !== $menu_item_type ) {
+				continue;
+			}
+
+			$old_menu_item_object_id = get_post_meta( $new_post_id, '_menu_item_object_id' )[ 0 ];
```
Comment (yossi-elementor) [inline core/app/modules/import-export/import.php:110]: Try to merge this query with the query on line 104.

File: core/app/modules/import-export/import.php [hunk 3]
```diff
@@ -26,7 +31,29 @@ final public function run() {
...
+
+		$imported_posts = $root_directory->run_import( $manifest_data );
+
+		remove_filter( 'elementor/document/save/data', [ $this, 'prevent_saving_elements_on_post_creation' ], 10 );
+
+		$this->map_old_new_post_ids = $this->map_old_new_post_ids( $imported_posts );
+
+		$this->save_elements_of_imported_posts();
+
+		$this->update_object_id_of_imported_menu_items();
```
Comment (yossi-elementor) [inline core/app/modules/import-export/import.php:44]: It's may be better to pass the array to the function and not reference it from within.

File: core/app/modules/import-export/module.php [hunk 4]
```diff
@@ -70,6 +70,7 @@ public function get_summary_titles() {
 		}
 
 		$post_types = get_post_types_by_support( 'elementor' );
+		$post_types[] = 'nav_menu_item';
```
Comment (yossi-elementor) [inline core/app/modules/import-export/module.php:73]: Make sure we really need this

File: core/utils/import-export/wp-exporter.php [hunk 5]
```diff
@@ -520,7 +520,7 @@ private function wxr_posts_list( array $post_ids ) {
 					$result .= $this->indent( 3 ) . '<link>' . esc_url( get_permalink() ) . '</link>' . PHP_EOL;
 					$result .= $this->indent( 3 ) . '<pubDate>' . mysql2date( 'D, d M Y H:i:s +0000', get_post_time( 'Y-m-d H:i:s', true ), false ) . '</pubDate>' . PHP_EOL;
 					$result .= $this->indent( 3 ) . '<dc:creator>' . $this->wxr_cdata( get_the_author_meta( 'login' ) ) . '</dc:creator>' . PHP_EOL;
-					$result .= $this->indent( 3 ) . '<guid isPermaLink="false">' . $this->wxr_cdata( get_the_author_meta( 'login' ) ) . '</guid>' . PHP_EOL;
+					$result .= $this->indent( 3 ) . '<guid isPermaLink="false">' . $this->wxr_cdata( $post->guid ) . '</guid>' . PHP_EOL;
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-exporter.php:523]: Please make sure with OA we're not breaking things.

File: core/utils/import-export/wp-import.php [hunk 6]
```diff
@@ -189,7 +189,7 @@ private function import( $file ) {
 			return $this->is_valid_meta_key( $key );
 		} );
 		add_filter( 'http_request_timeout', function () {
-			return self::DEFAULT_BUMP_REQUEST_TIMEOUT;
+			return static::DEFAULT_BUMP_REQUEST_TIMEOUT;
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:192]: Please revert to `self`

File: core/utils/import-export/wp-import.php [hunk 7]
```diff
@@ -326,7 +326,7 @@ private function set_author_mapping() {
 			return;
 		}
 
-		$create_users = apply_filters( 'import_allow_create_users', self::DEFAULT_ALLOW_CREATE_USERS );
+		$create_users = apply_filters( 'import_allow_create_users', static::DEFAULT_ALLOW_CREATE_USERS );
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:329]: Please revert to `self`

File: core/utils/import-export/wp-import.php [hunk 8]
```diff
@@ -1134,7 +1137,7 @@ private function fetch_remote_file( $url, $post ) {
 			return new WP_Error( 'import_file_error', esc_html__( 'Downloaded file has incorrect size', 'elementor' ) );
 		}
 
-		$max_size = (int) apply_filters( 'import_attachment_size_limit', self::DEFAULT_IMPORT_ATTACHMENT_SIZE_LIMIT );
+		$max_size = (int) apply_filters( 'import_attachment_size_limit', static::DEFAULT_IMPORT_ATTACHMENT_SIZE_LIMIT );
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:1140]: SAME

File: core/utils/import-export/wp-import.php [hunk 9]
```diff
@@ -1144,7 +1147,7 @@ private function fetch_remote_file( $url, $post ) {
 
 		// Override file name with Content-Disposition header value.
 		if ( ! empty( $headers['content-disposition'] ) ) {
-			$file_name_from_disposition = self::get_filename_from_disposition( (array) $headers['content-disposition'] );
+			$file_name_from_disposition = static::get_filename_from_disposition( (array) $headers['content-disposition'] );
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:1150]: SAME #2

File: core/utils/import-export/wp-import.php [hunk 10]
```diff
@@ -1153,7 +1156,7 @@ private function fetch_remote_file( $url, $post ) {
 		// Set file extension if missing.
 		$file_ext = pathinfo( $file_name, PATHINFO_EXTENSION );
 		if ( ! $file_ext && ! empty( $headers['content-type'] ) ) {
-			$extension = self::get_file_extension_by_mime_type( $headers['content-type'] );
+			$extension = static::get_file_extension_by_mime_type( $headers['content-type'] );
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:1159]: SAME #3

File: core/utils/import-export/wp-import.php [hunk 11]
```diff
@@ -1335,6 +1332,11 @@ public function run() {
 		return $this->output;
 	}
 
+	/**
+	 * @param $file
+	 * @param $args
+	 * @param $already_imported // Posts that already imported using different importer
```
Comment (yossi-elementor) [inline core/utils/import-export/wp-import.php:1338]: Please update the comment.

**Review by yossi-elementor — CHANGES_REQUESTED** [review]

**Comment by arielk** [comment]
@ManorHazaz Is this PR still relevant?

**Pull request closed without merging**